### PR TITLE
[v2.9] Bump docker-machine-driver-linode to v0.1.12 

### PR DESCRIPTION
**Pull request by ykim-1**
## Issues:
* https://github.com/linode/docker-machine-driver-linode/issues/131
* https://github.com/rancher/rancher/issues/46382
 
## Problem
bump docker-machine-driver-linode version to latest v0.1.12
 
## Solution
update dependancy to latest so that it reflects CVEs that are already fixed

### CODE REVIEW

**Review by pjbgf — DISMISSED** [review]
Thanks for raising this PR @ykim-1! :bow: 

Approving as per SURE-7072 and based on the [release changes](https://github.com/linode/docker-machine-driver-linode/releases/tag/v0.1.12).

**Comment by macedogm** [comment]
Can the following lines also be bumped to the same version, please?
- https://github.com/rancher/rancher/blob/release/v2.9/tests/v2/codecoverage/package/Dockerfile#L63
- https://github.com/rancher/rancher/blob/release/v2.9/pkg/data/management/machinedriver_data.go#L113

I honestly don't know if something else should be bumped too or not. Pinging Hostbusters and @snasovich for reviewing this also.

**Comment by ykim-1** [comment]
Apologies for the delay, completely forgot I had this PR open. I have addressed the comments, let me know if anything else should be updated

**Review by pjbgf — APPROVED** [review]

File: pkg/data/management/machinedriver_data.go [hunk 1]
```diff
@@ -110,7 +110,7 @@ func addMachineDrivers(management *config.ManagementContext) error {
 	if dl := os.Getenv("CATTLE_DEV_MODE"); dl != "" {
 		linodeBuiltin = isCommandAvailable("docker-machine-driver-linode")
 	}
-	if err := addMachineDriver(Linodedriver, "https://github.com/linode/docker-machine-driver-linode/releases/download/v0.1.11/docker-machine-driver-linode_linux-amd64.zip", "/assets/rancher-ui-driver-linode/component.js", "b31b6a504c59ee758d2dda83029fe4a85b3f5601e22dfa58700a5e6c8f450dc7", []string{"api.linode.com"}, linodeBuiltin, linodeBuiltin, false, management); err != nil {
+	if err := addMachineDriver(Linodedriver, "https://github.com/linode/docker-machine-driver-linode/releases/download/v0.1.12/docker-machine-driver-linode_linux-amd64.zip", "/assets/rancher-ui-driver-linode/component.js", "b31b6a504c59ee758d2dda83029fe4a85b3f5601e22dfa58700a5e6c8f450dc7", []string{"api.linode.com"}, linodeBuiltin, linodeBuiltin, false, management); err != nil {
```
Comment (snasovich) [inline pkg/data/management/machinedriver_data.go:113]: I see checksum wasn't updated the last time the driver was bumped (my bad missing it on https://github.com/rancher/rancher/pull/42021) but I see it was updated before (see https://github.com/rancher/rancher/issues/21838) and current checksum still matches one from version 0.1.8. Let's update it to match checksum from 0.1.12.

**Review by snasovich — CHANGES_REQUESTED** [review]

File: pkg/data/management/machinedriver_data.go (hunk 1 above)
Comment (ykim-1) [reply pkg/data/management/machinedriver_data.go]: Forgot to update the checksum, should be good to go now

**Review by snasovich — APPROVED** [review]

**Review by macedogm — APPROVED** [review]

**Review by macedogm — APPROVED** [review]

**Comment by ykim-1** [comment]
Thank you for the reviews @snasovich and @macedogm, would one of you guys be able to merge the PRs? I am blocked from merging due to `The base branch restricts merging to authorized users`

**Comment by macedogm** [comment]
@ykim-1 FYI the merge is being tracked in https://github.com/rancher/rancher/issues/46382.

**Pull request merged**